### PR TITLE
Improve formatting of multiline tokens

### PR DESCRIPTION
**Pull request by ooflorent**
Prior this change, multiline tokens such as `CommentBlock` or `TemplateLiteral` were not triggering _break mode_ formatting.

I also fixed the location reporting of some lint rules.

### CODE REVIEW

File: packages/@romejs/js-compiler/api/analyzeDependencies.test.ts [hunk 1]
```diff
@@ -27,208 +28,313 @@ async function testAnalyzeDeps(input: string, sourceType: ConstSourceType) {
...
   async (t) => {
-    await t.snapshot(await testAnalyzeDeps(`
-      exports.foo = 'bar';
-    `, 'module'));
+    await t.snapshot(
+      await testAnalyzeDeps(
+        dedent`
+          exports.foo = 'bar';
+        `,
+        'module',
```
Comment (ooflorent) [inline packages/@romejs/js-compiler/api/analyzeDependencies.test.ts:292]: I believe this test is wrong and this line should be changed to:
```suggestion
        'script',
```

File: packages/@romejs/js-compiler/lint/rules/noDupeArgs.test.md [hunk 2]
```diff
@@ -8,14 +8,14 @@
...
   ✖ Duplicate argument a in function definition
 
-  > 2 │   function hello(a, a) {
-      │                     ^
-    3 │     console.log("Hello);
-    4 │   }
+  > 1 │ function hello(a, a) {
+      │                   ^
+    2 │   //
+    3 │ }a
```
Comment (ooflorent) [inline packages/@romejs/js-compiler/lint/rules/noDupeArgs.test.md:18]: Strange reporting here. This is a bug in the codeframe.

File: packages/@romejs/js-compiler/lint/rules/noLabelVar.test.md [hunk 3]
```diff
@@ -8,14 +8,13 @@
...
 
   ✖ Labels should not be variable names
 
-    2 │       const x = "test";
-  > 3 │       x: const y = "test";
-      │       ^^^^^^^^^^^^^^^^^^^^
-    4 │       
+    1 │ const x = 'test';
+  > 2 │ x: const y = 'test';
+      │ ^^^^^^^^^^^^^^^^^^^^
```
Comment (ooflorent) [inline packages/@romejs/js-compiler/lint/rules/noLabelVar.test.md:17]: Initially I wanted to update the location of the diagnostic to `node.label` which would be better. However, this is a bug in the codeframe that discard to `^^^`…

Also, looking at the previous snapshot reveals a position issue. It seems that the `^^^` markers are placed according to the formatted output.

File: packages/@romejs/js-compiler/lint/rules/noDupeArgs.test.md (hunk 2 above)
Comment (sebmck) [reply packages/@romejs/js-compiler/lint/rules/noDupeArgs.test.md:18]: We have syntax highlighting that builds the source text from an array of tokens. I've seen it mess up sometimes but not sure what the cause is.

**Comment by sebmck** [comment]
Looks good! Could you resolve the conflict? Looks to have occured after merging the last PR. Happy to merge after.

**Pull request merged**